Metrics.go

```diff
@@ -19,10 +19,10 @@ type (
 		Histograms    map[string]prometheus.Histogram
 		Gauges        map[string]prometheus.Gauge
 		Logger        *logger.Logger
-		countMutex    *sync.Mutex
-		countVecMutex *sync.Mutex
-		histMutex     *sync.Mutex
-		gaugeMutex    *sync.Mutex
+		countMutex    *sync.RWMutex
```

Comment:
Well channels would for example make sense if we'd transfer control across GoRoutines, but in this case, we simply need to read from or write to a map (from concurrent routines) essentially guaranteeing that we instantiate each element in the map only once. Maps are not suited for concurrent access (even reads) which is why we must guard access via a Mutex. In this case, we use the `RWMutex` which allows us to have concurrent reads.